biosys/apps/main/api/serializers.py

```diff
@@ -66,6 +66,23 @@ def set_context(self, serializer_field):
         ]
     )
 
+    def update(self, instance, validated_data):
+        has_data = Record.objects.filter(dataset=instance).count() > 0
+        if has_data:
+            different_type = instance.type != validated_data.get('type')
+            different_data_package = instance.data_package != validated_data.get('data_package')
```

Comment:
Not hundred percent. I've tried. Some permutations work not other. I thought it should be enough for now knowing that user will probably change something instead of moving things around.